{{cookiecutter.repostory_name}}/devops/vultr_scripts/vultr-deploy.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+# deploy to list of IPs from `instances_ip.txt` (see `vultr-get-instances.py`)
+
+import subprocess
+from pathlib import Path
+
+pwd = Path(__file__).parent
+
+
+with open(pwd / "instances_ip.txt") as f:
+    ips = f.readlines()
+
+errs = []
+for ip in ips:
+    print('deploying to', ip)
+    try:
+        res = subprocess.Popen(
+            ["git", "push", f"root@{ip.strip()}:~/repos/{{ cookiecutter.django_project_name }}-central.git"],
+            env={
+                "GIT_SSH_COMMAND": "ssh -o ConnectTimeout=10  -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no"
+            },
+        ).communicate()
+    except subprocess.CalledProcessError:
+        errs.append(ip)
```

Comment:
Is there a known subset of errors that are common? Maybe a retry could be in place for these failing ones?

Reply:
only case I encountered is ssh timeout when the instance is unresponsive or I havent updated IPs after rotating some machines. No reason to retry these.